cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,496 @@
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		assert_ok!(PeopleRococoIdentity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			id.para.main.clone(),
+		));
+
+		assert_ok!(PeopleRococoIdentity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
+
+		// No events get triggered when calling set_sub_no_deposit
+
+		// No amount should be reserved as deposit amounts are set to 0.
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+		assert_eq!(reserved_bal, 0);
+		assert!(PeopleRococoIdentity::identity(&PeopleRococoSender::get()).is_some());
+		let (_, sub_accounts) =
+			<PeopleRococo as PeopleRococoPallet>::Identity::subs_of(&PeopleRococoSender::get());
+		assert_eq!(sub_accounts.len(), 1);
+	});
+}
+
+fn assert_reap_id_relay(total_deposit: u128, id: &Identity) {
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		let free_bal_before_reap = RococoBalances::free_balance(RococoRelaySender::get());
+		let reserved_balance = RococoBalances::reserved_balance(RococoRelaySender::get());
+
+		match id.relay.subs {
+			Subs::Zero => assert_eq!(reserved_balance, id_deposit_relaychain(&id.relay.main)),
+			_ => assert_eq!(reserved_balance, total_deposit),
+		}
+
+		assert_ok!(RococoIdentityMigrator::reap_identity(
+			RococoOrigin::root(),
+			RococoRelaySender::get()
+		));
+
+		let remote_deposit = match id.relay.subs {
+			Subs::Zero => calculate_remote_deposit(id.relay.main.encoded_size() as u32, 0),
+			Subs::One => calculate_remote_deposit(id.relay.main.encoded_size() as u32, 1),
+			Subs::Many(n) => calculate_remote_deposit(id.relay.main.encoded_size() as u32, n),
+		};
+
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(BalancesEvent::Unreserved { who, amount }) => {
+					who: *who == RococoRelaySender::get(),
+					amount: *amount == total_deposit,
+				},
+			]
+		);
+		assert!(PeopleRococoIdentity::identity(&RococoRelaySender::get()).is_none());
+		let (_, sub_accounts) = RococoIdentity::subs_of(&RococoRelaySender::get());
+		assert_eq!(sub_accounts.len(), 0);
+
+		let reserved_balance = RococoBalances::reserved_balance(RococoRelaySender::get());
+		// after reap reserved balance should be 0
+		assert_eq!(reserved_balance, 0);
+		let free_bal_after_reap = RococoBalances::free_balance(RococoRelaySender::get());
+
+		// free balance after reap should be greater than before reap
+		assert!(free_bal_after_reap > free_bal_before_reap);
+
+		match id.relay.subs {
+			Subs::Zero => {
+				assert_eq!(
+					free_bal_after_reap,
+					free_bal_before_reap + id_deposit_relaychain(&id.relay.main) - remote_deposit
+				);
+			},
+			_ => {
+				assert_eq!(
+					free_bal_after_reap,
+					free_bal_before_reap + total_deposit - remote_deposit
+				);
+			},
+		}
```

Comment:
This seems unnecessary. A variable like `total_deposit` should represent the total deposit.

Reply:
Yes, completely unnecessary!